service/service.go

```diff
@@ -120,7 +120,7 @@ func flatten(data map[string]interface{}, driverOptions *types.DriverOptions) {
 			driverOptions.StringSliceOptions[k] = &types.StringSlice{Value: v.([]string)}
 		case map[string]interface{}:
 			// hack for labels
-			if k == "labels" {
+			if k == "labels" || k == "resourceLabels" {
```

Comment:
@nathan-jenan-rancher I wonder if we can support map type in driverOptions, so that we don't need to extend the hancking list here?

Reply:
GRPC supports map types 
https://developers.google.com/protocol-buffers/docs/proto3#maps
So I don't see any reason why we couldn't do this.